src/System.Management.Automation/engine/CommandCompletion/CommandCompletion.LastWordFinder.cs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace System.Management.Automation
+{
+    public partial class CommandCompletion
+    {
+        /// <summary>
+        /// LastWordFinder implements the algorithm we use to search for the last word in a line of input taken from the console.
+        /// This class exists for legacy purposes only - V3 and forward uses a slightly different interface.
+        /// </summary>
+        private sealed class LastWordFinder
```

Comment:
The only use of this seems to be for legacy and should be removed

Reply:
I cloned your branch and removed the legacy code and builds fine, we'll have to look at the test failures (if any) with those changes

Reply:
Test failures in [build 111723](https://dev.azure.com/powershell/2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_build/results?buildId=111723) (e86ca2ae0106f85462a8af2b8438fabe83589a0a):
```log
 TabCompletion.User-overridden TabExpansion implementations.Override TabExpansion with function

Exception calling "CompleteInput" with "3" argument(s): "Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')"
```

```log
 TabCompletion.User-overridden TabExpansion implementations.Override TabExpansion with alias

Exception calling "CompleteInput" with "3" argument(s): "Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')"
```

I've reset to e86ca2ae0106f85462a8af2b8438fabe83589a0a and applied changes from review.